net/rpc/server_test.go

```diff
@@ -455,6 +469,22 @@ func TestServeRequest(t *testing.T) {
 	testServeRequest(t, newServer)
 }
 
+func TestServeRequestWithInterceptor(t *testing.T) {
+	a := 1
+	interceptor := ServerInterceptor(func(serviceMethod string, argv, replyv reflect.Value, handler func()) {
+		a = 2
```

Comment:
We should probably capture all of the values passed to this interceptor so that we can assert we got the values we expected.